pkg/utils/utils.go

```diff
@@ -220,39 +226,31 @@ func ReplaceAllContainerImageApply(img K8sImagesConfig, c *acorev1.ContainerAppl
 	case csmv1.Externalhealthmonitor:
 		*c.Image = img.Images.Externalhealthmonitor
 	case csmv1.Sdc:
-		*c.Image = img.Images.Sdcmonitor
+		*c.Image = img.Images.Sdc // DISCUSS: Why does this point to Sdcmonitor and not Sdc?
 	case csmv1.Sdcmonitor:
 		*c.Image = img.Images.Sdcmonitor
 	case string(csmv1.Resiliency):
 		*c.Image = img.Images.Podmon
 	}
 }
 
-// UpdateinitContainerApply -
-func UpdateinitContainerApply(initContainers []csmv1.ContainerTemplate, c *acorev1.ContainerApplyConfiguration) {
-	for _, init := range initContainers {
-		if *c.Name == init.Name {
-			if init.Image != "" {
-				*c.Image = string(init.Image)
-			}
-			if init.ImagePullPolicy != "" {
-				*c.ImagePullPolicy = init.ImagePullPolicy
-			}
-			emptyEnv := make([]corev1.EnvVar, 0)
-			c.Env = ReplaceAllApplyCustomEnvs(c.Env, emptyEnv, init.Envs)
-			c.Args = ReplaceAllArgs(c.Args, init.Args)
-
-		}
-	}
+// UpdateInitContainerApply -
+func UpdateInitContainerApply(initContainers []csmv1.ContainerTemplate, c *acorev1.ContainerApplyConfiguration) {
+	UpdateContainerApply(initContainers, c)
 }
 
 // ReplaceAllApplyCustomEnvs -
+// TODO: It seems like this function only acts on envs that are shared between
```

Comment:
Is this TODO needed?

Reply:
Mentioned in related https://github.com/dell/csm-operator/pull/785#discussion_r1847147990 

I have doubts that this function is behaving as intended and would like to examine the intent/behavior with others.